tests/flows/add_fee_amount.go

```diff
@@ -91,13 +98,19 @@ func AddFeeAmount(network network.Network) {
 	Expect(err).Should(BeNil())
 	Expect(delivered).Should(BeTrue())
 
-	// Check the relayer reward amount
-	amount, err :=
-		subnetAInfo.TeleporterMessenger.CheckRelayerRewardAmount(&bind.CallOpts{}, fundedAddress, mockTokenAddress)
+	// Check the updated relayer reward amount
+	expectedIncrease := new(big.Int).Add(initFeeAmount, additionalFeeAmount)
+	newRewardAmount, err := subnetAInfo.TeleporterMessenger.CheckRelayerRewardAmount(
+		&bind.CallOpts{},
+		receiveEvent.RewardRedeemer,
+		mockTokenAddress)
 	Expect(err).Should(BeNil())
-	Expect(amount).Should(Equal(additionalFeeAmount.Add(additionalFeeAmount, initFeeAmount)))
+	Expect(newRewardAmount).Should(Equal(new(big.Int).Add(initialRewardAmount, expectedIncrease)))
 
-	utils.RedeemRelayerRewardsAndConfirm(
-		ctx, subnetAInfo, mockToken, mockTokenAddress, fundedKey, amount,
-	)
+	// If the funded address is the one able to redeem the rewards, do so and check the reward amount is reset.
+	if fundedAddress == receiveEvent.RewardRedeemer {
```

Comment:
Could we use `network.SupportsIndependentRelaying` here? Perhaps renaming that method `FundedAddressIsRelayer` or similar would be more accurate.

Reply:
Same applies to this check in the other tests.

Reply:
I considered it, but I think this is actually the more accurate check. An implementation could theoretically support independent relaying but not be the only relayer for that network. In the event that a different relayer was the one to deliver the message, the rewards would still not be redeemable.

Reply:
Makes sense 👍